bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -534,6 +563,198 @@ impl Client {
         })))
     }
 
+    /// Subscribe to updates of global account data events.
+    ///
+    /// Be careful that only the most recent value can be observed. Subscribers
+    /// are notified when a new value is sent, but there is no guarantee that
+    /// they will see all values.
+    pub fn observe_account_data_event(
+        &self,
+        event_type: AccountDataEventType,
+        listener: Box<dyn AccountDataListener>,
+    ) -> Arc<TaskHandle> {
+        match event_type {
+            AccountDataEventType::Direct => {
+                // Using an Arc here is mandatory or else the subscriber will never trigger
+                let observer = Arc::new(
+                    self.inner
+                        .observe_events::<RumaGlobalAccountDataEvent<DirectEventContent>, ()>(),
+                );
+
+                Arc::new(TaskHandle::new(get_runtime_handle().spawn(async move {
+                    let mut subscriber = observer.subscribe();
+                    loop {
+                        if let Some(next) = subscriber.next().await {
+                            listener.on_change(next.0.into());
+                        }
+                    }
+                })))
+            }
```

Comment:
There's quite a bit of duplicated boilerplate between the various match arms. I couldn't find a good way to avoid this but maybe somebody with stronger rust-fu has a tip?

Reply:
How about a macro?

Something like:

```rust
($t:ty) => {{
    // Using an Arc here is mandatory or else the subscriber will never trigger
    let observer = Arc::new(self.inner.observe_events::<$t, ()>());

    Arc::new(TaskHandle::new(get_runtime_handle().spawn(async move {
        let mut subscriber = observer.subscribe();
        loop {
            if let Some(next) = subscriber.next().await {
                listener.on_change(next.0.into());
            }
        }
    })))
}};
```